client/src/pages/App.js

```diff
@@ -119,22 +121,32 @@ const App = ({ pageDispatchers, pageProps }) => {
   const history = useHistory()
   const routerLocation = useLocation()
   const { loading, error, data, refetch } = API.useApiQuery(GQL_GET_APP_DATA)
+
   const { done, result } = useBoilerplate({
     loading,
     error,
     pageProps,
     pageDispatchers
   })
+  const skip = done || error || !data
+  const appState = skip ? null : processData(data)
+  const [notifications, loadingNotifications] = useNotifications(
+    appState?.currentUser,
+    skip
+  )
+
   if (done) {
     return result
   }
+  if (loadingNotifications) {
+    // FIXME: need this otherwise, 404 tests breaks, how can we combine appData + notifications data and loading processes
+    return <div className="loader" />
+  }
 
-  if (error || !data) {
```

Comment:
if `error` exist, we can't reach here, `done` would take care of it above

Reply:
Indeed.